src/core_plugins/kibana/public/dashboard/selectors/index.js

```diff
@@ -1,98 +1,87 @@
-import {
-  getEmbeddableTitle as getEmbeddableTitleFromEmbeddables,
-  getEmbeddableEditUrl as getEmbeddableEditUrlFromEmbeddables,
-  getEmbeddableError as getEmbeddableErrorFromEmbeddables,
-  getEmbeddable as getEmbeddableFromEmbeddables,
-} from './embeddables';
-
-import {
-  getPanel as getPanelFromPanels,
-  getPanelType as getPanelTypeFromPanels
-} from './panels';
+/**
+ * @typedef {Object} ViewState
+ * @property {DashboardViewMode} viewMode
+ * @property {boolean} isFullScreenMode
+ * @property {string|undefined} maximizedPanelId
+ */
 
-import {
-  getViewMode as getViewModeFromView,
-  getFullScreenMode as getFullScreenModeFromView,
-  getMaximizedPanelId as getMaximizedPanelIdFromView
-} from './view';
+/**
+ * @typedef {Object} EmbeddableState
+ * @property {string} title
+ * @property {string} editUrl
+ * @property {string|object} error
+ */
 
 /**
  * @typedef {Object} DashboardState
- * @property {Object} PanelsState
- * @property {Object} EmbeddablesState
- * @property {Object} ViewState
+ * @property {Object.<string, PanelState>} panels
+ * @property {Object.<string, EmbeddableState>} embeddables
+ * @property {ViewState} view
  */
 
 /**
  * @param dashboard {DashboardState}
- * @return {PanelsState}
+ * @return {Object.<string, PanelState>}
  */
 export const getPanels = dashboard => dashboard.panels;
+
 /**
  * @param dashboard {DashboardState}
  * @param panelId {string}
  * @return {PanelState}
  */
-export const getPanel = (dashboard, panelId) => getPanelFromPanels(getPanels(dashboard), panelId);
+export const getPanel = (dashboard, panelId) => getPanels(dashboard)[panelId];
```

Comment:
Honestly, I think it's fine to do:
```
export const getPanel = (dashboard, panelId) => dashboard.panels[panelId];
```

Reply:
I left it like this because I'm using the `getPanels` function in the one level higher up:

`export const getPanels = state => DashboardSelectors.getPanels(getDashboard(state));`

So I figured if I had the function already I should use it.  Otherwise I'd have to change the one higher up to

`export const getPanels = state => getDashboard(state).panels;`

The higher up level is a pain because it's going to get really big if we throw all core app state in there (as mentioned in the other file).

I could maybe do something like you were showing and just call getDashboard(state).panels in here but that breaks my current contract where the "state" at each level matches the file name (e.g. not the root state but the dashboard state here).

I think I'd just as well leave it like this for now and sit on future improvements for a little bit.

Reply:
Though... I'm using it inconsistently anyway since in a lot of places I'm not using selectors at the root level, but at the dashboard level and grabbing state via a param like ({ dashboard })... so I already have that logic scattered about...

Reply:
Ah, I'm sticking with my original thought to leave it as is for now.  I have a lot of PRs in progress that this is going to create conflicts for, so I'd prefer to do these improvement incrementally, once I'm sure which styles and patterns I like/dislike.

Reply:
Perfectly fine!